### DISCUSSION TITLE
【Matcha】更好的本地开发体验，与你的机器人直接进行交互

### DISCUSSION THREAD

**Discussion by A-kirami** [nonebot/discussions#351] [应用]
## 背景
在 Bot 或者插件的开发过程中，有时需要测试实际效果，通常我们有两种选择。

### 单元测试

编写单元测试，确保代码能够通过检验。

它的优点是测试效率高，缺点有：

1. 编写测试用例需要时间精力，有时可能测试代码比业务代码还多（
2. 不够直观，无法直接看到完整效果

### 真实测试

在真实的平台上进行测试，能展现最真实、最直观的效果，所见即所得。

但是它却也有一些问题：

1. 登陆平台需要一个账号
2. 平台具有某些限制
3. 可能的风控或者冻结

### 还有其他选择吗？

当然！现在你有了一个新选择——[Matcha](https://github.com/A-kirami/matcha) 。

## 介绍

<p align="center">
  <a href="https://github.com/A-kirami/matcha" target="__blank">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://raw.githubusercontent.com/A-kirami/matcha/main/public/preview-dark.webp">
      <source media="(prefers-color-scheme: light)" srcset="[.](https://raw.githubusercontent.com/A-kirami/matcha/main)/public/preview-light.webp">
      <img src="https://raw.githubusercontent.com/A-kirami/matcha/main/public/preview-light.webp" alt="Matcha - Preview" width="100%" />
    </picture>
  </a>
</p>

Matcha 不需要编写复杂的测试用例，也不需要真实的平台账号。

它能够直接与 Bot 进行模拟聊天交互，降低开发者的调试与测试的负担，从而更有效率的专注于功能开发。

### 多媒体消息支持

支持文本、图片、音频、视频消息

![image](https://user-images.githubusercontent.com/66513481/226441811-42f39dfb-4363-4afa-921d-d680e2eda979.png)

### 切换显示内容

可以切换展示原始事件内容

![image](https://user-images.githubusercontent.com/66513481/226442401-95453312-dcc5-4222-808f-4260b352651f.png)

## 支持

OneBot v11 协议大部分支持

OneBot v12 协议完全支持

## 项目地址：https://github.com/A-kirami/matcha

**Comment by mnixry** [comment]
超人
